flink-kubernetes/src/test/java/org/apache/flink/kubernetes/kubeclient/decorators/KerberosMountDecoratorTest.java

```diff
@@ -0,0 +1,147 @@
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThat;
+
+/**
+ * General tests for the {@link KerberosMountDecoratorTest}.
+ */
+public class KerberosMountDecoratorTest extends KubernetesJobManagerTestBase {
+
+	private KerberosMountDecorator kerberosMountDecorator;
+
+	@Override
+	protected void setupFlinkConfig() {
+		super.setupFlinkConfig();
+
+		flinkConfig.set(SecurityOptions.KERBEROS_LOGIN_KEYTAB, kerberosDir.toString() + "/" + KEYTAB_FILE);
+		flinkConfig.set(SecurityOptions.KERBEROS_LOGIN_PRINCIPAL, "test");
+		flinkConfig.set(KubernetesConfigOptions.KERBEROS_KRB5_PATH, kerberosDir.toString() + "/" + KRB5_CONF_FILE);
+	}
+
+	@Override
+	protected void onSetup() throws Exception {
+		super.onSetup();
+
+		generateKerberosFileItems();
+
+		this.kerberosMountDecorator = new KerberosMountDecorator(kubernetesJobManagerParameters);
+	}
+
+	@Test
+	public void testWhetherPodOrContainerIsDecorated() {
+		final FlinkPod resultFlinkPod = kerberosMountDecorator.decorateFlinkPod(baseFlinkPod);
+		assertNotEquals(baseFlinkPod.getPod(), resultFlinkPod.getPod());
+		assertNotEquals(baseFlinkPod.getMainContainer(), resultFlinkPod.getMainContainer());
+	}
+
+	@Test
+	public void testConfigMap() throws IOException {
+		final List<HasMetadata> additionalResources = kerberosMountDecorator.buildAccompanyingKubernetesResources();
+		assertEquals(2, additionalResources.size());
+
+		final ConfigMap resultConfigMap = (ConfigMap) additionalResources
+			.stream()
+			.filter(x -> x instanceof ConfigMap)
+			.collect(Collectors.toList()).get(0);
+
+		assertEquals(Constants.API_VERSION, resultConfigMap.getApiVersion());
+
+		assertEquals(KerberosMountDecorator.getKerberosKrb5confConfigMapName(CLUSTER_ID),
+			resultConfigMap.getMetadata().getName());
+
+		Map<String, String> resultDatas = resultConfigMap.getData();
+		assertThat(resultDatas.size(), is(1));
+		assertEquals("some conf", resultDatas.get(KRB5_CONF_FILE));
+	}
+
+	@Test
+	public void testDecoratedFlinkPod() {
+		final FlinkPod resultFlinkPod = kerberosMountDecorator.decorateFlinkPod(baseFlinkPod);
+		assertEquals(2, resultFlinkPod.getPod().getSpec().getVolumes().size());
+
+		final List<Volume> expectedVolumes = Arrays.asList(
+			new VolumeBuilder()
+				.withName(Constants.KERBEROS_KEYTAB_VOLUME)
+				.withNewSecret()
+				.withSecretName(KerberosMountDecorator.getKerberosKeytabSecretName(CLUSTER_ID))
+				.endSecret()
+				.build(),
+			new VolumeBuilder()
+				.withName(Constants.KERBEROS_KRB5CONF_VOLUME)
+				.withNewConfigMap()
+				.withName(KerberosMountDecorator.getKerberosKrb5confConfigMapName(CLUSTER_ID))
+				.withItems(new KeyToPathBuilder()
+					.withKey(KRB5_CONF_FILE)
+					.withPath(KRB5_CONF_FILE)
+					.build())
+				.endConfigMap()
+				.build());
+		assertEquals(expectedVolumes, resultFlinkPod.getPod().getSpec().getVolumes());
+	}
```

Comment:
I think this test case is not needed. It is already covered by `testDecoratedFlinkContainer`. Despite it is required to mount the volumes on the pods before mounting them on the containers, from the contract we only cares about accessing the resources in container.